process/factory/shard/vmContainerFactory.go

```diff
@@ -60,18 +71,55 @@ func NewVMContainerFactory(args ArgVMContainerFactory) (*vmContainerFactory, err
 	cryptoHook := hooks.NewVMCryptoHook()
 	builtinFunctions := blockChainHookImpl.GetBuiltinFunctionNames()
 
-	return &vmContainerFactory{
-		config:                         args.Config,
-		blockChainHookImpl:             blockChainHookImpl,
-		cryptoHook:                     cryptoHook,
-		blockGasLimit:                  args.BlockGasLimit,
-		gasSchedule:                    args.GasSchedule,
-		builtinFunctions:               builtinFunctions,
-		deployEnableEpoch:              args.DeployEnableEpoch,
-		aheadOfTimeGasUsageEnableEpoch: args.AheadOfTimeGasUsageEnableEpoch,
-		arwenV3EnableEpoch:             args.ArwenV3EnableEpoch,
-		ArwenESDTFunctionsEnableEpoch:  args.ArwenESDTFunctionsEnableEpoch,
-	}, nil
+	factory := &vmContainerFactory{
+		config:                               args.Config,
+		blockChainHookImpl:                   blockChainHookImpl,
+		cryptoHook:                           cryptoHook,
+		blockGasLimit:                        args.BlockGasLimit,
+		gasSchedule:                          args.GasSchedule,
+		builtinFunctions:                     builtinFunctions,
+		epochNotifier:                        args.EpochNotifier,
+		deployEnableEpoch:                    args.DeployEnableEpoch,
+		aheadOfTimeGasUsageEnableEpoch:       args.AheadOfTimeGasUsageEnableEpoch,
+		arwenV3EnableEpoch:                   args.ArwenV3EnableEpoch,
+		ArwenESDTFunctionsEnableEpoch:        args.ArwenESDTFunctionsEnableEpoch,
+		ArwenGasManagementRewriteEnableEpoch: args.ArwenGasManagementRewriteEnableEpoch,
+		container:                            nil,
+	}
+
+	factory.arwenVersions, err = factory.prepareVersions(args.Config.ArwenVersions)
+
+	return factory, nil
+}
+
+func (vmf *vmContainerFactory) prepareVersions(versionsByEpochs []config.VersionByEpochs) ([]config.VersionByEpochs, error) {
```

Comment:
this is actually a sort and verification - thus may rename. ?

Reply:
Split in two functions with better naming.